ntropy_sdk/income_check.py

```diff
@@ -151,36 +162,77 @@ def from_income_groups(cls, income_groups: List[IncomeGroup]):
             passive_income=round(passive_income_amount, 2),
             possible_income=round(undetermined_amount, 2),
             earned_income_sources=sorted(set(earned_income_sources)),
-            passive_income_sources=sorted(set(passive_income_source)),
+            passive_income_sources=sorted(set(passive_income_sources)),
             possible_income_sources=sorted(set(undetermined_sources)),
         )
 
 
-class IncomeReport:
+class IncomeReport(list):
     def __init__(self, income_groups: List[IncomeGroup]):
+        """Parameters
+        ----------
+        income_groups : List[IncomeGroup]
+            A list of IncomeGroup objects.
+        """
+        super().__init__(income_groups)
         self.income_groups = income_groups
 
-    @classmethod
-    def from_dicts(cls, income_report: List[Dict[str, Any]]):
-        income_groups = sorted(
-            [IncomeGroup.from_dict(d) for d in income_report],
-            key=lambda x: float(x.total_amount),
-            reverse=True,
-        )
-        return cls(income_groups=income_groups)
-
     def to_df(self) -> Any:
         try:
             import pandas as pd
         except ImportError:
             raise RuntimeError("pandas is not installed")
-        return pd.DataFrame(self.to_json())
+        df = pd.DataFrame(self.json())
+        df.transaction_ids = df.transaction_ids.apply(lambda x: len(x))
+        df = df.rename({"transaction_ids": "# transactions"})
+        return df
 
-    def to_json(self) -> List[Dict[str, Any]]:
+    def json(self) -> List[Dict[str, Any]]:
         return [ig.dict() for ig in self.income_groups]
 
     def summarize(self) -> IncomeSummary:
         return IncomeSummary.from_income_groups(self.income_groups)
 
+    def _repr_df(self) -> Any:
+        try:
+            import pandas as pd
+        except ImportError:
+            raise RuntimeError("pandas is not installed")
```

Comment:
we need a fallback here instead?

Reply:
the fallback happens on the main method. Should be fine like this. if someone try to use this method directly without pandas, it will get a runtime error. But for the __repr__ method, it will fallback to using strings only